src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx

```diff
@@ -102,123 +107,188 @@ const NodeSelector = memo(({ schemata, height }: NodeSelectorProps) => {
         return [...byCategories.values()].flat().filter((n) => favorites.has(n.schemaId));
     }, [byCategories, favorites]);
 
+    const [showCollapseButtons, setShowCollapseButtons] = useState(false);
+
+    const defaultIndex = [['', favoriteNodes]].concat([...byCategories]).map((_, index) => index);
+    const [accordionIndex, setAccordionIndex] = useState<ExpandedIndex>(defaultIndex);
+
+    const toggleAccordion = () => {
+        if (typeof accordionIndex !== 'number' && accordionIndex.length === 0)
```

Comment:
Please extract this condition out into a variable. It's not trivial, and you're using it further below again.